doc/dev/sharp_bits.rst

```diff
@@ -0,0 +1,944 @@
+
+            @cond(y < 0.5)
+            def cond_gate():
+                qml.CRX(y * jnp.exp(- y ** 2), wires=[0, 1])
+
+            cond_gate()
+
+            return y * 2
+
+        for_loop(0, 3, step=1)(measurement_loop)(x)
+        return qml.expval(qml.PauliZ(0))
+
+>>> print(qml.draw(circuit)(0.3))
+0: ──RX(0.30)─╭●─╭●─────────RX(0.60)─╭●──RX(1.20)─╭●─┤  <Z>
+1: ──RY(0.09)─╰X─╰RX(0.27)──RY(0.36)─╰X──RY(1.44)─╰X─┤
+
+At the moment, additional PennyLane `circuit inspection functions
+<https://docs.pennylane.ai/en/stable/introduction/inspecting_circuits.html>`__
+are not supported with Catalyst.
+
+Classical control debugging
+---------------------------
+
+.. note::
+
+    See our AutoGraph guide for converting native Python control flow
+    to QJIT compatible control.
+
+There are various constraints and restrictions that should be kept in mind
+when working with classical control in Catalyst.
+
+- The return values of all branches of :func:`~.cond` must be the same type.
+  Returning different types, or ommitting a return value in one branch (e.g.,
+  returning ``None``) but not in others will result in an error.
+
+  >>> @qjit
+  ... def f(x: float):
+  ...     @cond(x > 1.5)
+  ...     def cond_fn():
+  ...         return x ** 2  # float
+  ...     @cond_fn.otherwise
+  ...     def else_branch():
+  ...         return 6  # int
+  ...     return cond_fn()
+  TypeError: Conditional requires consistent return types across all branches, got:
+  - Branch at index 0: [ShapedArray(float64[], weak_type=True)]
+  - Branch at index 1: [ShapedArray(int64[], weak_type=True)]
+  Please specify an else branch if none was specified.
+  >>> @qjit
+  ... def f(x: float):
+  ...     @cond(x > 1.5)
+  ...     def cond_fn():
+  ...         return x ** 2  # float
+  ...     @cond_fn.otherwise
+  ...     def else_branch():
+  ...         return 6.  # float
+  ...     return cond_fn()
+  >>> f(1.5)
+  array(6.)
+
+- Similarly, the else (``my_cond_fn.otherwise``) may be omitted **as long as
+  other branches do not return any values**. If other branches do return values,
+  the else branch must be specified.
+
+  >>> @qjit
+  ... def f(x: float):
+  ...     @cond(x > 1.5)
+  ...     def cond_fn():
+  ...         return x ** 2
+  ...     return cond_fn()
+  TypeError: Conditional requires consistent return types across all branches, got:
+  - Branch at index 0: [ShapedArray(float64[], weak_type=True)]
+  - Branch at index 1: []
+  Please specify an else branch if none was specified.
+
+  >>> @qjit
+  ... def f(x: float):
+  ...     @cond(x > 1.5)
+  ...     def cond_fn():
+  ...         return x ** 2
+  ...     @cond_fn.otherwise
+  ...     def else_branch():
+  ...         return x
+  ...     return cond_fn()
+  >>> f(1.6)
+  array(2.56)
+
+- Finally, a reminder that conditional functions provided to :func:`~.cond` cannot
+  accept any arguments.
+
+
+Compatibility with PennyLane transforms
+---------------------------------------
```

Comment:
@erick-xanadu this is based on the current `main` branch, so will need to be updated when your PR is merged